index.bs

```diff
@@ -222,27 +245,36 @@ To <dfn>sanitize an element</dfn> named |element|, run these steps:
 4. if |name| is in |config|'s [=element drop list=] return 'drop'.
 5. if |name| is contained in the built-in [=default element block list=] return 'block'.
 6. if |name| is in |config|'s [=element block list=] return 'block'.
-7. if |config| has a non-empty [=element allow list=] and |name| is not in |config|'s [=element allow list=] return 'block'
-8. return 'keep'
+7. if |config| has a non-empty [=element allow list=] and |name| is not in |config|'s [=element allow list=] return 'block'.
+8. [=list/iterate|for each=] |attr| in |element|'s [=Element/attribute list=]:
+  1. call [=sanitize an attribute=] with |attr|'s name and |element|'s tag name.
```

Comment:
I _think_ you need to use something like element's "local name".. We also need to think about namespaces here. Maybe just leave a `// TODO` comment for now

Reply:
Yes, TODO (or Issue) is totally warranted. An alternative might be to use the constructor name (HTMLScriptElement), since that'd help us with custom elements.

Reply:
Done.

(Used "local name" instead of tag name, and added an Issue: statement that we will need to revisit this.)

Reply:
I'd really like to get this closed, as it's been out for a while. My preference would still be to land this as-is.


I do wonder about alternatives. I think the use case that you want to block <img src=""> but not other src= attributes is real. But maybe we can make it easier for people that don't want this, and specify that the attribute lists can be either a list/set - in which case they apply universally - or a dictionary - in which case they select element/attribute combos. (Not sure if that's idiomatic IDL, though.)